### PR TITLE
Revert "Revert D30752939: [pytorch][PR] nvfuser update" (#65137)

### PR DESCRIPTION
**Pull request by jjsjann123**
cherry-picked code cleaning from: https://github.com/pytorch/pytorch/pull/65137

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/runtime/helpers.cu [hunk 1]
```diff
@@ -121,11 +121,11 @@ __device__ int64_t where(bool c, int64_t a, int64_t b) {
   return c ? a : b;
 }
 
-__device__ double randLike(Philox& rnd) {
+__device__ double randLike(Philox rnd) {
```
Comment (naoyam) [inline torch/csrc/jit/codegen/cuda/runtime/helpers.cu]: This reverts a recent fix. I'm guessing there's a merge conflict?

File: torch/csrc/jit/codegen/cuda/index_compute.cpp [hunk 2]
```diff
@@ -260,7 +260,7 @@ void updateHaloInfoForReference(
     auto consumer_root_axis = *consumer_it;
     auto root_axis_info =
         gpu_lower->haloInfo().getRootAxisInfo(consumer_root_axis);
-    if (root_axis_info.width() == 0) {
+    if (root_axis_info.width() == nullptr) {
```
Comment (naoyam) [inline torch/csrc/jit/codegen/cuda/index_compute.cpp]: Thanks for catching this. The original code is not correct. It should be:

```
if (!root_axis_info.width()->isZeroInt()) {
```

File: torch/csrc/jit/codegen/cuda/scheduler/registry.h [hunk 3]
```diff
@@ -189,13 +189,13 @@ class TORCH_CUDA_CU_API SchedulerEntry {
   const bool has_reduction_param_;
 
   //! Reduction parameters if applicable
-  ReductionParams rparams_;
+  ReductionParams rparams_ = ReductionParams();
```
Comment (naoyam) [inline torch/csrc/jit/codegen/cuda/scheduler/registry.h]: nit: isn't the constructor called by default?

Comment (jjsjann123) [reply torch/csrc/jit/codegen/cuda/scheduler/registry.h]: I think so. maybe there's something with linter at the time of PR.
Let's ditch the change then. Hopefully linter would be quieter next time. :crossed_fingers:

**Review by naoyam — APPROVED** [review]
LGTM.

File: torch/csrc/jit/codegen/cuda/index_compute.cpp (hunk 2 above)
Comment (jjsjann123) [reply torch/csrc/jit/codegen/cuda/index_compute.cpp]: Should this be `if (root_axis_info.width()->isZeroInt()) {` instead?
The other one seems to be breaking cpp tests.

Comment (naoyam) [reply torch/csrc/jit/codegen/cuda/index_compute.cpp]: My bad, you're right.

**Pull request merged**